informer/src/informer_service/informer_service_handler.rs

```diff
@@ -98,3 +121,68 @@ impl InformerService {
         }
     }
 }
+
+pub struct BlockCreatedEventGenerator {
+    client: Arc<Client>,
+    rate_limiter: Limiter,
+    send_status: ColdEventSendStatus,
+}
+
+impl BlockCreatedEventGenerator {
+    pub fn new(client: Arc<Client>) -> Self {
+        Self {
+            client,
+            rate_limiter: ratelimit::Builder::new().quantum(1).interval(Duration::new(0, 5000)).build(),
+
+            send_status: ColdEventSendStatus::InProgress,
+        }
+    }
+
+    pub fn run(mut self, connection: Connection, mut favorite_block_number: u64) {
+        println!("I am here 3");
+        let rt = Runtime::new().unwrap();
+        rt.spawn(async move {
+            task::spawn(async move {
+                loop {
+                    let event = self.gen(favorite_block_number);
+                    connection.cold_notification(&event);
+                    self.rate_limiter.wait();
```

Comment:
You should not call `blocking` function in an async function.
The `rate_limiter.wait` is a blocking call. You must not use it.
You can implement a simple rate limiter on your own. Or you can search for a rate limiter that can be used with async functions.